src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -546,6 +546,173 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+static void computeGCMProduct(MacroAssembler* masm,
+                              VectorRegister vLowerH, VectorRegister vH, VectorRegister vHigherH,
+                              VectorRegister vConstC2, VectorRegister vZero, VectorRegister vState,
+                              VectorRegister vTmp4, VectorRegister vTmp5, VectorRegister vTmp6,
+                              VectorRegister vTmp7, VectorRegister vTmp8, VectorRegister vTmp9,
+                              VectorRegister vTmp10, VectorRegister vTmp11, Register data) {
+    assert(masm != nullptr, "MacroAssembler pointer is null");
+    masm->vxor(vH, vH, vState);
+    masm->vpmsumd(vTmp4, vLowerH, vH);     // L : Lower Half of subkey H
+    masm->vpmsumd(vTmp5, vTmp11, vH);      // M : Combined halves of subkey H
+    masm->vpmsumd(vTmp6, vHigherH, vH);    // H : Higher Half of subkey H
+    masm->vpmsumd(vTmp7, vTmp4, vConstC2); // Reduction
+    masm->vsldoi(vTmp8, vTmp5, vZero, 8);  // mL : Extract the lower 64 bits of M
+    masm->vsldoi(vTmp9, vZero, vTmp5, 8);  // mH : Extract the higher 64 bits of M
+    masm->vxor(vTmp4, vTmp4, vTmp8);       // LL + LL : Partial result for lower half
+    masm->vxor(vTmp6, vTmp6, vTmp9);       // HH + HH : Partial result for upper half
+    masm->vsldoi(vTmp4, vTmp4, vTmp4, 8);  // Swap
+    masm->vxor(vTmp4, vTmp4, vTmp7);       // Reduction using constant
+    masm->vsldoi(vTmp10, vTmp4, vTmp4, 8); // Swap
+    masm->vpmsumd(vTmp4, vTmp4, vConstC2); // Reduction
+    masm->vxor(vTmp10, vTmp10, vTmp6);     // Combine reduced Low & High products
+    masm->vxor(vState, vTmp4, vTmp10);
+    masm->addi(data, data, 16);
```

Comment:
I think incrementing the data pointer fits better into the loop instead of this helper function.
Don't forget that hotspot uses 2 spaces indentation!